pkg/server/server.go

```diff
@@ -1055,17 +1055,23 @@ func (s *Server) ServerID() uint64 {
 // StoreInternalSession implements SessionManager interface.
 // @param addr	The address of a session.session struct variable
 func (s *Server) StoreInternalSession(se any) {
+	var count int
 	s.sessionMapMutex.Lock()
 	s.internalSessions[se] = struct{}{}
+	count = len(s.internalSessions)
 	s.sessionMapMutex.Unlock()
+	metrics.InternalSessions.Set(float64(count))
```

Comment:
I hope we can move it before `Unlock`. So that the last executed `InternalSessions.Set` is always the latest value guarded by `sessionMapMutex` when `StoreInternalSession` is concurrency called.

Reply:
It's not need to be very accurate, at the cost of performance.
Allocation after lock / or visit network after lock ... are all costly.  We can avoid those as much as possible.

This value is constantly changing, and prometheus update the data every 15s (if I remember correctly)
So the accuracy does not make much difference -- you can get hundreds or thousands values in the 15s interval, and all of them are discarded (not collected)

@lance6716

Reply:
It's just an atomic operation, I don't think it will affect performance.